### PR TITLE
More fixups to module resolution

### PR DESCRIPTION
**Pull request by DanielRosenwasser**

### CODE REVIEW

File: pages/Module Resolution.md [hunk 1]
```diff
@@ -44,20 +44,20 @@ Use non-relative paths when importing any of your external dependnecies.
 
 ## Module Resolution Strategies
 
-Use `--moduleResolution` to specify the module resolution strategy.
-Possible values are [Classic](#classic) and [Node](#node).
-The default if not specified is [Node](#node).
+There are two possible module resolution strategies [Node](#node) and [Classic](#classic).
```
Comment (sandersn) [inline pages/Module Resolution.md]: need colon after 'strategies'

File: pages/Module Resolution.md [hunk 2]
```diff
@@ -44,20 +44,20 @@ Use non-relative paths when importing any of your external dependnecies.
...
+You can use the `--moduleResolution` flag to specify the module resolution strategy.
+Unless otherwise specified, [Node](#node) is the default.
```
Comment (sandersn) [inline pages/Module Resolution.md]: I like the old sentence better. No extraposed prepositional phrase.

File: pages/Module Resolution.md [hunk 3]
```diff
@@ -65,88 +65,113 @@ For example:
...
-* `/src/node_modules/moduleB/index.ts`
-* `/src/node_modules/moduleB/index.tsx`
-* `/node_modules/moduleB.ts`
-* `/node_modules/moduleB.tsx`
-* `/node_modules/moduleB.d.ts`
-* `/node_modules/moduleB/package.json` look for `"typings"` property
-* `/node_modules/moduleB/index.ts`
-* `/node_modules/moduleB/index.tsx`
-* `/node_modules/moduleB/index.d.ts`
+1. `/foo/src/node_modules/moduleB.ts`
```
Comment (sandersn) [inline pages/Module Resolution.md]: I talked to Mohamed about this -- he decided to cut the nesting to one to keep the example short, even though it's not necessarily clear whether the walk upwards visits intermediate nodes. People can read the Node page to find that out, though. Our sample should try to be short and clear, so I'd go back to the original single-level example.

File: pages/Module Resolution.md [hunk 4]
```diff
@@ -65,88 +65,113 @@ For example:
...
 
 To understand what the steps the TS compiler will follow, it is important to shed some light on Node.js modules.
-Node.js supports multiple ways to resolve the target of a call to `require`.
-A relative module name in Node.js can resolve to:
+Traditionally, imports in Node are performed by calling a function named `require`.
+The behavior Node takes will differ depending on if `require` is given a relative path or a non-relative path.
 
-* A file
+Non-relative paths are fairly straightforward.
+As an example, let's consider a file located at `/foo/src/moduleA.js`, which contains the import `var x = require("./moduleB");`
```
Comment (sandersn) [inline pages/Module Resolution.md]: Remove some words:
"As an example, consider the file `/src/moduleA.js' with the import `var x ...`."

File: pages/Module Resolution.md [hunk 5]
```diff
@@ -65,88 +65,113 @@ For example:
...
+Node will end up trying to resolve that import in the following order:
```
Comment (sandersn) [inline pages/Module Resolution.md]: Remove some words: "Node will resolve that import ..."

File: pages/Module Resolution.md [hunk 6]
```diff
@@ -65,88 +65,113 @@ For example:
...
 
- For instance in `/src/moduleA.js` a require statement such as `var x = require("./moduleB");` refers to `/src/moduleB.js`.
- See [Node.js file modules documentation](https://nodejs.org/api/modules.html#modules_file_modules) for more details.
+1. As the file named `/foo/src/moduleB.js`, if one exists.
```
Comment (sandersn) [inline pages/Module Resolution.md]: if it exists

File: pages/Module Resolution.md [hunk 7]
```diff
@@ -65,88 +65,113 @@ For example:
...
 
-* A folder, with an implicit "main" module named `index.js`
+2. As the folder `/foo/src/moduleB` if it contains a file named `package.json`.
```
Comment (sandersn) [inline pages/Module Resolution.md]: I would combine the next line with this one: "... contains a file named `package.json` that specifies a `"main"` module.

File: pages/Module Resolution.md [hunk 8]
```diff
@@ -65,88 +65,113 @@ For example:
...
+   In order for resolution to succeed here, it will need to specify its `"main"` module.
 
- For instance in `/src/moduleA.js` and require statement `var x = require("./moduleB");` refers to `/src/moduleB/index.js`.
- See [Node.js folder modules documentation](https://nodejs.org/api/modules.html#modules_folders_as_modules) for more details.
+   In our example, if Node found the file `/foo/src/moduleB/package.json` and it contained something like `{ "main": "lib/mainModule.js" }`, then Node will refer to `/src/moduleB/lib/mainModule.js`.
```
Comment (sandersn) [inline pages/Module Resolution.md]: Remove words: "... the file `/src/moduleB/package.json` containing `{` ..."

File: pages/Module Resolution.md [hunk 9]
```diff
@@ -65,88 +65,113 @@ For example:
...
 #### How Node.js resolves modules
 
 To understand what the steps the TS compiler will follow, it is important to shed some light on Node.js modules.
-Node.js supports multiple ways to resolve the target of a call to `require`.
-A relative module name in Node.js can resolve to:
+Traditionally, imports in Node are performed by calling a function named `require`.
+The behavior Node takes will differ depending on if `require` is given a relative path or a non-relative path.
 
-* A file
+Non-relative paths are fairly straightforward.
```
Comment (sandersn) [inline pages/Module Resolution.md]: Relative paths, right? Or is the next section wrong? I'm getting confused between relative and non-relative. It's relative: "./foo" vs non-relative: "foo", right?

**Comment by sandersn** [comment]
On the whole, I like the original better. It's less wordy and the examples are simpler. Links to the Node documentation should suffice to explain Node's *exact* algorithm.

File: pages/Module Resolution.md [hunk 10]
```diff
@@ -65,88 +65,113 @@ For example:
...
+3. As the folder `/foo/src/moduleB` if it contains a file named `index.js`.
+   That file is implicitly considered that folder's "main" module.
 
- For instance in `/src/moduleA.js`, a require statement `var x = require("./moduleB");`, and a `/src/moduleB/package.json` file `{ "main": "lib/mainModule.js" }` refers to `/src/moduleB/lib/mainModule.js`.
+You can read more about this in Node's documentation on [file modules](https://nodejs.org/api/modules.html#modules_file_modules) and [folder modules](https://nodejs.org/api/modules.html#modules_folders_as_modules).
 
-Moreover, a require for a [non-relative module name](#relative-vs-non-relative-module-imports) is resolved from a special folder named `node_modules`.
-`node_modules` can be on the same level as the current file, or higher up in the directory chain.
-Node.js will *walk up* the directory chain, inspecting the contents of `node_modules` looking for the module to load.
+However, resolution for a [non-relative module name](#relative-vs-non-relative-module-imports) is performed little differently.
```
Comment (sandersn) [inline pages/Module Resolution.md]: remove 'little'

File: pages/Module Resolution.md [hunk 11]
```diff
@@ -65,88 +65,113 @@ For example:
...
+Node will look for your modules in special folders named `node_modules`.
```
Comment (sandersn) [inline pages/Module Resolution.md]: we should just use present tense everywhere when describing the algorithm: "Node looks", "Node walks", etc.

Comment (mhegazy) [inline pages/Module Resolution.md]: Node.js

File: pages/Module Resolution.md [hunk 12]
```diff
@@ -65,88 +65,113 @@ For example:
...
+A `node_modules` folder can be on the same level as the current file, or higher up in the directory chain.
+Node will walk up the directory chain, looking through each `node_modules` until it finds the module you tried to load.
 
-Following up our example above, in `/src/moduleA.js`, and a require statement `var x = require("moduleB");` will result in the following attempts:
+Following up our example above, consider if `/foo/src/moduleA.js` instead used a non-relative path and had the import `var x = require("moduleB");`.
+Node would then try to resolve `moduleB` to each of the locations until one worked.
```
Comment (mhegazy) [inline pages/Module Resolution.md]: Node.js

File: pages/Module Resolution.md [hunk 13]
```diff
@@ -65,88 +65,113 @@ For example:
...
+4. `/foo/node_modules/moduleB.js`
+5. `/foo/node_modules/moduleB/package.json` (if it specifies a `"main"` property)
+6. `/foo/node_modules/moduleB/index.js`
+   <br /><br />
+7. `/node_modules/moduleB.js`
+8. `/node_modules/moduleB/package.json` (if it specifies a `"main"` property)
+9. `/node_modules/moduleB/index.js`
 
-See [Node.js loading modules from `node_modules` documentation](https://nodejs.org/api/modules.html#modules_loading_from_node_modules_folders) for more details.
+Notice that Node jumped up a directory in steps (4) and (7).
```
Comment (mhegazy) [inline pages/Module Resolution.md]: Node.js

**Comment by mhegazy** [comment]
just use `Node.js` everywhere you have `Node` now for consistency.

File: pages/Module Resolution.md (hunk 1 above)
Comment (mhegazy) [inline pages/Module Resolution.md]: done,

File: pages/Module Resolution.md (hunk 2 above)
Comment (mhegazy) [inline pages/Module Resolution.md]: done.

File: pages/Module Resolution.md (hunk 5 above)
Comment (mhegazy) [inline pages/Module Resolution.md]: done.

File: pages/Module Resolution.md (hunk 6 above)
Comment (mhegazy) [inline pages/Module Resolution.md]: done

File: pages/Module Resolution.md (hunk 7 above)
Comment (mhegazy) [inline pages/Module Resolution.md]: done.

File: pages/Module Resolution.md (hunk 8 above)
Comment (mhegazy) [inline pages/Module Resolution.md]: done.

File: pages/Module Resolution.md (hunk 9 above)
Comment (mhegazy) [inline pages/Module Resolution.md]: done.

File: pages/Module Resolution.md (hunk 10 above)
Comment (mhegazy) [inline pages/Module Resolution.md]: done.

**Pull request merged**